src/solvers/dg_triangles/dg.jl

```diff
@@ -0,0 +1,287 @@
+end
+
+# interpolates from solution coefficients to face quadrature points
+function prolong2interfaces!(cache, u, mesh::AbstractMeshData, equations, 
+               surface_integral, dg::DG{<:RefElemData})
+  rd = dg.basis    
+  @unpack u_face_values = cache
+  StructArrays.foreachfield(mul_by!(rd.Vf), u_face_values, u)
+end
+
+function create_cache(mesh::VertexMappedMesh, equations, dg::DG, RealT, uEltype) where {DG <: DGWeakForm{2}}
+
+  rd = dg.basis
+  md = mesh.md
+  # mass matrix, differentiation matrices
+  @unpack M, Dr, Ds = rd
+  # volume quadrature weights, volume interpolation matrix
+  @unpack wq, Vq = rd 
+
+  # ∫f(u) * dv/dx_i = (Vq*Dr)'*diagm(wq)*(rxJ.*f(Vq*u)) + (Vq*Ds)'*diagm(wq)*(sxJ.*f(Vq*u))
+  invMQrTrW = -M\((Vq*Dr)'*diagm(wq))
+  invMQsTrW = -M\((Vq*Ds)'*diagm(wq))
+
+  nvars = nvariables(equations)
+
+  # storage for volume quadrature values, face quadrature values, flux values
+  u_values = StructArray{SVector{nvars, RealT}}(ntuple(_->zeros(rd.Nq, md.num_elements), nvars))
+  u_face_values = StructArray{SVector{nvars, RealT}}(ntuple(_->zeros(rd.Nfq, md.num_elements), nvars))
+  flux_face_values = similar(u_face_values)
+
+  # local storage for fluxes
+  flux_values = StructArray{SVector{nvars,RealT}}(ntuple(_->zeros(rd.Nq), nvars))
+  
+  return (; md, invMQrTrW, invMQsTrW, invJ=inv.(md.J),
+      u_values, flux_values, u_face_values, flux_face_values)
+end
+
+function calc_volume_integral!(du,u::StructArray, volume_integral::VolumeIntegralWeakForm,
+                 mesh::VertexMappedMesh, equations, dg::DG{<:RefElemData{2}}, cache)
+
+  rd = dg.basis  
+  md = mesh.md
+  @unpack invMQrTrW, invMQsTrW, u_values, flux_values = cache
+  @unpack rxJ, sxJ, ryJ, syJ = md # geometric terms
+
+  # interpolate to quadrature points
+  StructArrays.foreachfield(mul_by!(rd.Vq), u_values, u)
+
+  # todo: dispatch on curved/non-curved mesh types. This only works for affine meshes (accessing rxJ[1,e],...)
+  for e in Base.OneTo(md.num_elements)
+    flux_values .= flux.(view(u_values,:,e), 1, equations)
+    StructArrays.foreachfield(mul_by_accum!(invMQrTrW, rxJ[1,e]), view(du,:,e), flux_values)
+    StructArrays.foreachfield(mul_by_accum!(invMQsTrW, sxJ[1,e]), view(du,:,e), flux_values)
+
+    flux_values .= flux.(view(u_values,:,e),2,equations)
+    StructArrays.foreachfield(mul_by_accum!(invMQrTrW, ryJ[1,e]), view(du,:,e), flux_values)
+    StructArrays.foreachfield(mul_by_accum!(invMQsTrW, syJ[1,e]), view(du,:,e), flux_values)
+  end
+end
+
+# calc_interface_flux!(cache, dg.surface_integral, mesh, equations, dg)
+function calc_interface_flux!(cache, surface_integral::SurfaceIntegralWeakForm, 
+                mesh::VertexMappedMesh, equations, 
+                dg::DG{<:RefElemData{2}}) 
+
+  @unpack surface_flux = surface_integral
+  md = mesh.md
+  @unpack mapM, mapP, nxJ, nyJ, sJ = md 
+  @unpack u_face_values, flux_face_values = cache 
+
+  num_face_nodes = length(u_face_values)
+  for i in Base.OneTo(num_face_nodes)
```

Comment:
```suggestion
  for face in Base.OneTo(num_face_nodes)
```
We often try to use a bit more suggestive names for loop indices like this, e.g.
- `for element in eachelement(...)`
- `for boundary in achboundary(...)`
- `for interface in eachinterface(...)`
- `for mortar in eachmortar(...)`

It could be nice to use some similar style for these new solvers

Reply:
Can do. I'll go ahead and use `mesh`, `equations`, `dg`, `cache` as per the discussion in https://github.com/trixi-framework/Trixi.jl/issues/661#issuecomment-866914977.

Reply:
Actually, I think I'll leave off `equations` from the signature. I feel like that doesn't make much sense to include.